src/plot_api/plot_api.js

```diff
@@ -419,7 +416,16 @@ function opaqueSetBackground(gd, bgColor) {
 }
 
 function setPlotContext(gd, config) {
-    if(!gd._context) gd._context = Lib.extendDeep({}, defaultConfig);
+    if(!gd._context) {
+        gd._context = Lib.extendDeep({}, defaultConfig);
+
+        // stash <base> href, used to make robust clipPath URLs
+        var base = d3.select('base');
+        gd._context.baseUrl = base.size() && base.attr('href') ?
+            window.location.href.split('#')[0] :
+            '';
```

Comment:
This isn't something we will ever expose as a setting to users, right? I'd make it `_baseUrl` then.